### ISSUE TITLE
[BUG] Nesting level too deep on PearlHandler

### ISSUE THREAD

**Issue by CaptainDuck** [Olybear9/Mavoric#60]
## Description
The crash happens whenever you throw enderpearls.

Crashdump:
```
PocketMine-MP Crash Dump Wed Apr 22 06:54:11 UTC 2020

Error: Nesting level too deep - recursive dependency?
File: plugins/Mavoric.phar/src/Bavfalcon9/Mavoric/Core/Handlers/PearlHandler
Line: 83
Type: E_ERROR

THIS CRASH WAS CAUSED BY A PLUGIN
BAD PLUGIN: Mavoric v1.0.5

Code:
[74] 
[75]     /**
[76]      * Clears throws from memory for performance reasons.
[77]      * @return void
[78]      */
[79]     public function purge(): void {
[80]         $purged = 0;
[81]         foreach ($this->throws as &$throw) {
[82]             if ($throw->getLandingTime() + 10 <= time()) {
[83]                 unset($this->throws[array_search($throw, $this->throws)]);
[84]                 $purged++;
[85]             }
[86]         }
[87]         return;
[88]     }
[89] 
[90]     /**
[91]      * Returns a list of throws from a player
[92]      * @param String $player - Player name
[93]      * @return PearlThrow[] - Array of throws

Backtrace:

PocketMine-MP version: 3.12.0 [Protocol 390]
Git commit: 77d45bf11674a1360f58e32a14a3b09f423d860c
uname -a: Linux 6f061c12-5b23-45fb-a92d-45635b991a8b 3.10.0-957.27.2.el7.x86_64 #1 SMP Mon Jul 29 17:46:05 UTC 2019 x86_64
PHP Version: 7.3.12
Zend version: 3.3.12
OS : Linux, linux
```
**Additional context**

Thank you!

**Comment by Olybear9** [comment]
The origin of this issue is due to the passing of `$throw` as a reference.
This issue has been fixed on versions v1.0.6 and above.

**Comment by CaptainDuck** [comment]
All right, thanks! There is another issue tho, NoClip also detects if you are inside a portal, should it he exempted?
